web/src/todo_widget.ts

```diff
@@ -210,24 +295,41 @@ export class TaskData {
         },
     };
 
-    handle_event(sender_id, data) {
+    handle_event(sender_id: number, data: InboundData): void {
         const type = data.type;
-        if (this.handle[type] && this.handle[type].inbound) {
+        if (this.handle[type]?.inbound !== undefined) {
             this.handle[type].inbound(sender_id, data);
         } else {
             blueslip.warn(`todo widget: unknown inbound type: ${type}`);
         }
     }
 }
 
-export function activate({$elem, callback, extra_data, message}) {
+export function activate({
+    $elem,
+    callback,
+    extra_data,
+    message,
+}: {
+    $elem: JQuery;
+    callback: (
+        data: NewTaskOutboundData | StrikeOutboundData | NewTaskListTitleOutboundData | undefined,
+    ) => void;
+    extra_data: {
+        task_list_title?: string;
+        tasks?: {task: string; desc: string}[];
+    } | null;
```

Comment:
The type for this has already been defined in widgetize.ts, which you can move to this file now.

In general, try to base this work off `poll_widget.ts` since these are pretty similar files, and ensure to test thoroughly by sending and interacting with a variety of poll messages.

Reply:
> The type for this has already been defined in widgetize.ts, which you can move to this file now.

I tried using the type, but it didn't work out quite well. It resulted in a lot of errors.

> In general, try to base this work off `poll_widget.ts` since these are pretty similar files, and ensure to test thoroughly by sending and interacting with a variety of poll messages.

Yes, I have based this migration on `poll_widget.ts`. Also, I have tested the Todo Widget extensively, by trying all the different ways the to-do widget can be interacted with.

Reply:
> I tried using the type, but it didn't work out quite well. It resulted in a lot of errors.

Could you be more specific about the errors? If the type that had been working so far is resulting in errors, it's highly likely the problem is with the rest of the code.